e2e_playwright/shared/dataframe_utils.py

```diff
@@ -0,0 +1,221 @@
+) -> None:
+    """Click on the middle of the specified cell.
+
+    Parameters
+    ----------
+    dataframe_element : Locator
+        The dataframe element to click on.
+
+    row_pos : int
+        The row number to click on. Starts at 0 with the header row.
+
+    col_pos : int
+        The column number to click on. Starts with 0 with the first column.
+        If has_row_marker_col is True, the first column is the row marker column.
+
+    column_width : "small" | "medium" | "large"
+        The shared width setting of all columns. Can be "small", "medium" or "large".
+        This needs to be enforced in the dataframe via column config.
+
+    has_row_marker_col : bool
+        Whether the dataframe has a row marker column (used when row selections are activated).
+
+    double_click : bool
+        Whether to double click on the cell.
+    """
+    column_middle_width_px, row_middle_height_px = calc_middle_cell_position(
+        row_pos, col_pos, column_width, has_row_marker_col
+    )
+    position = {"x": column_middle_width_px, "y": row_middle_height_px}
+
+    if double_click:
+        dataframe_element.dblclick(position=position)
+    else:
+        dataframe_element.click(position=position)
+
+
+def select_row(
+    dataframe_element: Locator,
+    row_pos: int,
+    column_width: Literal["small"] | Literal["medium"] | Literal["large"] = "small",
+) -> None:
+    """Select the specified row in the dataframe.
+
+    This expects row selections to be activated.
+
+    Parameters
+    ----------
+
+    dataframe_element : Locator
+        The dataframe element to select the row in.
+
+    row_pos : int
+        The row number to select. Starts at 0 with the header row.
+
+    column_width : "small" | "medium" | "large"
+        The shared width setting of all columns. Can be "small", "medium" or "large".
+        This needs to be enforced in the dataframe via column config.
+    """
+    click_on_cell(dataframe_element, row_pos, 0, column_width, has_row_marker_col=True)
+
+
+def sort_column(
+    dataframe_element: Locator,
+    col_pos: int,
+    column_width: Literal["small"] | Literal["medium"] | Literal["large"] = "small",
+    has_row_marker_col: bool = False,
+) -> None:
+    """Sort the specified column in the dataframe.
+
+    Parameters
+    ----------
+
+    dataframe_element : Locator
+        The dataframe element to select the column in.
+
+    col_pos : int
+        The column number to select. Starts with 0 with the first column.
+        If has_row_marker_col is True, the first column is the row marker column.
+
+    column_width : "small" | "medium" | "large"
+        The shared width setting of all columns. Can be "small", "medium" or "large".
+        This needs to be enforced in the dataframe via column config.
+
+    has_row_marker_col : bool
+        Whether the dataframe has a row marker column (used when row selections are activated).
+    """
+    click_on_cell(
+        dataframe_element,
+        0,
```

Comment:
nit: it probably makes sense to add a comment that `0` is the header row or maybe better
```python
header_row_index = 0
  click_on_cell(
        dataframe_element,
        header_row_index,
...
```
to avoid using a magic number